dataquality/integrations/torch_semantic_segmentation.py

```diff
@@ -476,14 +477,18 @@ def upload_contours_split(self, split: str) -> None:
         local_contour_path = f"{project_path}/{config.current_run_id}/{split}/contours"
 
         files = os.listdir(local_contour_path)
-        all_contours = {}
-        for file in files:
-            with open(f"{local_contour_path}/{file}") as f:
-                contours = json.load(f)
+        image_to_contours_map: Dict = defaultdict(dict)
+        for fname in files:
+            with open(f"{local_contour_path}/{fname}") as f:
+                # Dict image_id -> contours for a single polygon in the image
+                image_to_contours_single_polygon = json.load(f)
+                image_id = list(image_to_contours_single_polygon.keys())[0]
```

Comment:
Feels a bit fragile. Might be nice to validate that we had:
a. A valid dict.
b. A dict with >1 keys.

```suggestion
                
                image_id = list(image_to_contours_single_polygon.keys())[0] if image_to_contours_single_polygon and len(image_to_contours_single_polygon.keys()) > 0
```